### PR TITLE
feat(share-summary): Add share topic button to meeting summary

### PR DESCRIPTION
**Pull request by igorlesnenko**
Partially solves: #8232 #8234

**Demo**

https://www.loom.com/share/1302d78b31494ff8a02a20d6f8790a4d

image

**In this PR**

- Added `share topic` to the email summary (works both in the email and the website)
- Button opens a modal (Added routing and modal placeholder)

**Not included**

- Modal UI
- I have some scroll issue when modal is opened, going to look into it in next PR (Please let me know if you know what is going on)

**How to test**

- Add `shareSummary` org-level feature flag
- Run a retro and have some topics
- Click on `Share topic` button, see modal is opened and routing works fine
- Click on `Share topic` button from email, see modal is opened and routing works fine

### CODE REVIEW

**Review by nickoferrall — CHANGES_REQUESTED** [review]
When I end the meeting, I'm getting the following error in the console:
```
10:19:44 1|GraphQL Executor | TypeError: Cannot destructure property 'history' of '(0 , hooks_useRouter\_WEBPACK_IMPORTED_MODULE_5__.default)(...)' as it is undefined.
```
I also don't receive the email confirmation. When I remove the feature flag and end the meeting, I do receive the email confirmation.

**Review by nickoferrall — CHANGES_REQUESTED** [review]
When I end the meeting, I'm getting the following error in the console:
```
10:19:44 1|GraphQL Executor | TypeError: Cannot destructure property 'history' of '(0 , hooks_useRouter\_WEBPACK_IMPORTED_MODULE_5__.default)(...)' as it is undefined.
```
I also don't receive the email confirmation. When I remove the feature flag and end the meeting, I do receive the email confirmation.

**Review by nickoferrall — APPROVED** [review]
LGTM! Nice job getting it working from the email too

File: packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/ShareTopic.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  if (isEmail) {
+    return (
+      <AnchorIfEmail isEmail={true} href={href} style={style} title={label}>
+        {label}
+      </AnchorIfEmail>
+    )
+  }
+
+  // Avoid calling hooks when doing SSR
+  /* eslint-disable react-hooks/rules-of-hooks */
```
[the remark this answers is not in the document]
Comment (igorlesnenko) [reply packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/ShareTopic.tsx:47]: Was not sure about it, as I will have to create 3 components just to workaround this case

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
Scrolling behaves weird when cancelling and then reopening the dialog after having scrolled
Peek 2023-06-14 14-58

**Comment by igorlesnenko** [comment]
@Dschoordsch I made a partial fixes for this. Scroll to top issue is related to routing. Was not able to find a quick solution yet. I probably need to save scroll position or show popup using `useState`. Can I look into in in some next PRs?

**Comment by igorlesnenko** [comment]
@Dschoordsch fixed by changing approach for showing modal window from browser

File: packages/client/hooks/useModal.ts [hunk 2]
```diff
@@ -19,7 +19,8 @@ const useModal = (options: Options = {}) => {
     id,
     parentId,
     onOpen,
-    onClose
+    onClose,
+    allowScroll: false
```
Comment (Dschoordsch) [inline packages/client/hooks/useModal.ts:23]: -1 Scrolling is broken on unrelated modals with your pr.
master:
master
this pr:
pr
Not sure if it's related to this line and I stopped checking after I found the first issue.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

**Pull request merged**